rollup/da_syncer/blob_client/blob_scan_client.go

```diff
@@ -0,0 +1,112 @@
+package blob_client
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+)
+
+type BlobScanClient struct {
+	client      *http.Client
+	apiEndpoint string
+}
+
+func NewBlobScanClient(apiEndpoint string) *BlobScanClient {
+	return &BlobScanClient{
+		client:      http.DefaultClient,
+		apiEndpoint: apiEndpoint,
+	}
+}
+
+func (c *BlobScanClient) GetBlobByVersionedHash(ctx context.Context, versionedHash common.Hash) (*kzg4844.Blob, error) {
+	// blobscan api docs https://api.blobscan.com/#/blobs/blob-getByBlobId
+	path, err := url.JoinPath(c.apiEndpoint, versionedHash.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to join path, err: %w", err)
+	}
+	req, err := http.NewRequestWithContext(ctx, "GET", path, nil)
```

Comment:
why we use `http.NewRequestWithContext(ctx, "GET", path, nil)` for `BlobScanClient`

but `http.Get(path)` for `BlockNativeClient`?

Reply:
because need to set header for this request